### PR TITLE
nikd01-cv04

### PR DESCRIPTION
**Pull request by dmitrynikolaev08**
https://eso.vse.cz/~nikd01/cv04/

### CODE REVIEW

File: www/nikd01/cv04/index.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
+<?php
+require_once './utils/utils.php';
+
+$name = $email = $password = $confirm_password = '';
+$filename = __DIR__ . '/database/users.db';
+$errors = [];
+
+if ($_SERVER["REQUEST_METHOD"] == "POST") {
+    $name = trim(filter_input(INPUT_POST, 'name', FILTER_SANITIZE_STRING));
```
Comment (nvbach91) [inline www/nikd01/cv04/index.php:9]: image

File: www/nikd01/cv04/utils/utils.php [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    // Encrypt the password before storing it
+    $hashed_password = password_hash($password, PASSWORD_DEFAULT);
+
+    // Prepare the user data as a CSV string
+    $user_record = implode(";", [$name, $email, $hashed_password]) . PHP_EOL;
+
+    // Append the user record to the file
+    file_put_contents($filename, $user_record, FILE_APPEND);
```
Comment (nvbach91) [inline www/nikd01/cv04/utils/utils.php:61]: po uspesne registraci neni zaznam uzivatele pridan https://eso.vse.cz/~nikd01/cv04/admin/users.php

**Review by nvbach91 — CHANGES_REQUESTED** [review]

File: www/nikd01/cv04/index.php (hunk 1 above)
Comment (dmitrynikolaev08) [reply www/nikd01/cv04/index.php]: This issue should be resolved now

File: www/nikd01/cv04/utils/utils.php (hunk 2 above)
Comment (dmitrynikolaev08) [reply www/nikd01/cv04/utils/utils.php:61]: now it should be okay - it wasn't the code related issue, i just didn't set the writing rights for the "database.db" file on the server

**Review by nvbach91 — APPROVED** [review]

**Pull request merged**